src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/GenericComponent.tsx

```diff
@@ -24,19 +21,20 @@ export interface IGenericComponentProps {
   dataModelBindings: IDataModelBindings;
 }
 
-export const GenericComponent = (props: IGenericComponentProps) => {
+export const getFormDataForComponent = (formData: any) => {
+  return formData ? formData : '';
+}
 
-  const GetFormDataSelector = makeGetFormDataSelector();
-  const GetLayoutElementSelector = makeGetLayoutElement();
+export const GenericComponent = (props: IGenericComponentProps) => {
   const GetHiddenSelector = makeGetHidden();
   const GetFocusSelector = makeGetFocus();
 
   const dataModel: IDataModelFieldElement[] = useSelector((state: IRuntimeState) => state.formDataModel.dataModel);
-  const formData: IFormData = useSelector((state: IRuntimeState) => GetFormDataSelector(state, props));
+  const formData: IFormData = useSelector((state: IRuntimeState) => getFormDataForComponent(state.formData.formData[props.dataModelBindings.simpleBinding]), shallowEqual);
```

Comment:
Does this work for advanced components which does not have a simpleBinding?

Reply:
wups, good point! Added support for advanced components.